[Wallet] Visual refinement 2

aaronmgdr opened the pull request:
### Description

1 set an explicit weight of normal on our regular font as without one in places where a third part component set a default weight it would not be overridden. example the drawer item component. 

2. use small not regular for the `PhoneNumbeWithEmoji` (Cory confirmed this was desired in both places it is used)

3. fix extra space at top of screens caused by safe area view. also

4. place exchange confirmation view's scroll view instead of the safe area view instead of other way to match how we have other screens set. 

5. hamburger in top bar was not centered vertically and was too far to the right. 




### Tested

![Screen Shot 2020-07-16 at 1 41 45 PM](https://user-images.githubusercontent.com/3814795/87731864-53916400-c780-11ea-9743-83ab41edc61e.png)
![Screen Shot 2020-07-16 at 3 02 25 PM](https://user-images.githubusercontent.com/3814795/87731865-5429fa80-c780-11ea-8e04-d1e5a29e3d94.png)
![Screen Shot 2020-07-16 at 3 02 33 PM](https://user-images.githubusercontent.com/3814795/87731866-5429fa80-c780-11ea-9290-eb6122acba4d.png)
![Screen Shot 2020-07-16 at 3 02 38 PM](https://user-images.githubusercontent.com/3814795/87731868-5429fa80-c780-11ea-8f04-6ee7b13755a4.png)
![Screen Shot 2020-07-16 at 3 02 46 PM](https://user-images.githubusercontent.com/3814795/87731869-54c29100-c780-11ea-8c4b-6dc809421b39.png)
![Screen Shot 2020-07-16 at 3 02 52 PM](https://user-images.githubusercontent.com/3814795/87731871-54c29100-c780-11ea-809f-f9e3d91594e4.png)
<img width="390" alt="Screen Shot 2020-07-16 at 3 51 44 PM" src="https://user-images.githubusercontent.com/3814795/87731872-54c29100-c780-11ea-893f-52546d5ae538.png">

### Related issues

- Fixes a few bugs on the wallet design bug list

Commit:
set explicit weight so that it superceeds third party set weight

Commit:
make number small all places

Commit:
fix double copy

Commit:
fix safe area padding on several screens to only apply to bottom edge

Commit:
fix hamburger alignment and top bar vertical size. use hitslop. instead of padding.

Commit:
Fixes : Too much top margin on:
All secondary Settings screens (Edit Profile, Language, Currency)
Send/Request Review screens
Buy/Sell CELO Review screens

Commit:
snapshots -- verified

Commit:
JUST set normal in this spot to avoid changing every snapshot we have

Commit:
revert, i was wrong this isnt causing double paste

aaronmgdr commented on packages/mobile/src/navigator/DrawerNavigator.tsx:
```diff
@@ -185,7 +185,7 @@ export default function DrawerNavigator() {
       drawerContent={drawerContent}
       backBehavior={'initialRoute'}
       drawerContentOptions={{
-        labelStyle: [fontStyles.regular, { marginLeft: -20 }],
```
without setting weight to normal we get 500 set by react-navigation.

note the margin -20 was here before

aaronmgdr replied:
i also considered setting theis globaly, but the snapshot diff was crazy big so i figured lets be conservative

jeanregisser commented on packages/mobile/src/exchange/ExchangeConfirmationCard.tsx:
```diff
@@ -101,8 +101,8 @@ export default function ExchangeConfirmationCard(props: Props) {
-      </SafeAreaView>
-    </ScrollView>
+      </ScrollView>
+    </SafeAreaView>
```
`SafeAreaView` should generally be inside `ScrollView` so it displays nicely on devices with a notch.
Any reason not to do that here?

aaronmgdr replied:
ah ha I got confused and thought it was the other way

aaronmgdr replied:
Looking at the code I see 2 places where SafeAreaView is inside the scroll view and  about 10+ where it is outside.

jeanregisser replied:
Yes when I initially added support for iOS, I put all of them outside as a shortcut.
In the long term most of them are meant to be inside, except when there's a fixed button area at the bottom of the screen.

jeanregisser commented on packages/react-components/styles/fonts.v2.tsx:
```diff
@@ -79,7 +79,7 @@ const fontStyles = StyleSheet.create({
   },
   large: standards.large,
-  regular: standards.regular,
+  regular: { ...standards.regular },
```
Why is this needed?

aaronmgdr replied:
ah leftover from when i tried adding font weight normal here.

jeanregisser reviewed (commented):
Thanks for these!! 👍 🎉 
Left some very small questions.

Commit:
revert safe scroll order, and more
spent a good amount of time trying to find a way to just type useNavigation to recognize that it is inside a Drawer but no luck.

Commit:
📸

jeanregisser reviewed (approved):
Great 👍

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/valora-sparkly

Commit:
📸
